File: app/src/main/java/mb00/android/codehub/ui/user/view/UserActivity.java
package mb00.android.codehub.ui.user.view;

import android.os.Bundle;

import mb00.android.codehub.R;
import mb00.android.codehub.data.BundleKeys;
import mb00.android.codehub.data.PreferenceKeys;
import mb00.android.codehub.databinding.ActivityUserBinding;
import mb00.android.codehub.ui.base.view.BaseDrawerActivity;
import mb00.android.codehub.ui.user.adapter.UserAdapter;
import mb00.android.codehub.ui.user.adapter.UserFragmentPagerAdapter;
import mb00.android.codehub.ui.user.viewmodel.UserViewModel;

/**
 * Launched from {@link UserAdapter} if user in RecyclerView is clicked
 * Immediately launches {@link UserFragmentPagerAdapter}
 */

public class UserActivity extends BaseDrawerActivity<ActivityUserBinding, UserViewModel> {

    private Bundle userBundle;
    private String userName;

    @Override
    protected int layout() {
        return R.layout.activity_user;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getExtras();
        initToolbar();
        initViewPager();
    }

    private void getExtras() {
        userBundle = getIntent().getExtras();
        userName = userBundle.getString(BundleKeys.USER_NAME);

        if (userName == null) {
            userName = getSharedPreferences(PreferenceKeys.PREFERENCES, MODE_PRIVATE)
                    .getString(PreferenceKeys.USER_NAME, "");
        }
    }

    private void initToolbar() {
        getBinding().userToolbarTextView.setText(userName);
    }

    private void initViewPager() {
        UserFragmentPagerAdapter userPagerAdapter = new UserFragmentPagerAdapter(getSupportFragmentManager(), this, userBundle);

        getBinding().userViewPager.setAdapter(userPagerAdapter);
        getBinding().userViewPager.setOffscreenPageLimit(7);
        getBinding().userViewPager.setPageMargin(4);
        getBinding().userViewPager.setPageMarginDrawable(R.color.grey);
        getBinding().userTabLayout.setupWithViewPager(getBinding().userViewPager);
        getBinding().userViewPager.setCurrentItem(userBundle.getInt(BundleKeys.VIEW_PAGER_POSITION));
    }

}

File: app/src/main/java/mb00/android/codehub/ui/user/view/UserActivity.kt
package mb00.android.codehub.ui.user.view

import android.content.Context
import android.os.Bundle
import mb00.android.codehub.R
import mb00.android.codehub.data.BundleKeys
import mb00.android.codehub.data.PreferenceKeys
import mb00.android.codehub.databinding.ActivityUserBinding
import mb00.android.codehub.ui.base.view.BaseDrawerActivity
import mb00.android.codehub.ui.user.adapter.UserAdapter
import mb00.android.codehub.ui.user.adapter.UserFragmentPagerAdapter
import mb00.android.codehub.ui.user.viewmodel.UserViewModel

/**
 * Launched from [UserAdapter] if user in RecyclerView is clicked
 * Immediately launches [UserFragmentPagerAdapter]
 */

class UserActivity : BaseDrawerActivity<ActivityUserBinding, UserViewModel>() {

    private var userBundle: Bundle? = null
    private var authHeader: String? = ""
    private var userName: String? = ""

    override fun layout(): Int {
        return R.layout.activity_user
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        getExtras()
        initToolbar()
        initViewPager()
    }

    private fun getExtras() {
        userBundle = intent.extras
        userName = userBundle?.getString(BundleKeys.USER_NAME)

        val preferences = getSharedPreferences(PreferenceKeys.PREFERENCES, Context.MODE_PRIVATE)
        authHeader = preferences.getString(PreferenceKeys.AUTH_HEADER, "")
        userName = getSharedPreferences(PreferenceKeys.PREFERENCES, Context.MODE_PRIVATE)
                .getString(PreferenceKeys.USER_NAME, "")
    }

    private fun initToolbar() {
        binding.userToolbarTextView.text = userName
    }

    private fun initViewPager() {
        val userPagerAdapter = UserFragmentPagerAdapter(supportFragmentManager, this, userBundle)

        binding.userViewPager.adapter = userPagerAdapter
        binding.userViewPager.offscreenPageLimit = 7
        binding.userViewPager.pageMargin = 4
        binding.userViewPager.setPageMarginDrawable(R.color.grey)
        binding.userTabLayout.setupWithViewPager(binding.userViewPager)

        userBundle?.let { binding.userViewPager.currentItem = userBundle?.getInt(BundleKeys.VIEW_PAGER_POSITION) ?: 0 }
    }

}
